packages/validators/src/schemas/componentDefinition.ts

```diff
@@ -22,14 +22,120 @@ export const ComponentDefinitionSchema = z.object({
   id: DefinitionPropertyKeySchema,
   variables: z.record(
     DefinitionPropertyKeySchema,
-    z.object({
-      // TODO - extend with definition of validations and defaultValue
-      displayName: z.string().optional(),
-      type: DefinitionPropertyTypeSchema,
-      description: z.string().optional(),
-      group: z.string().optional(),
-    }),
+    z
+      .object({
+        // TODO - extend with definition of validations and defaultValue
+        displayName: z.string().optional(),
+        type: DefinitionPropertyTypeSchema,
+        description: z.string().optional(),
+        group: z.string().optional(),
+        defaultValue: z.union([z.string(), z.number(), z.undefined(), z.any()]).optional(),
+      })
+      .superRefine((val, ctx) => {
```

Comment:
the majority of the relevant changes to validation are here in this superRefine method

Reply:
@elylucas did you consider using `z.or` instead of custom validators? I'm wondering whether a static validation would be better to handle, e.g. when later creating OpenAPI specs out of this (see the ongoing OpenAPI project that already supports zod and other validation libraries).

Reply:
I had not, do you have an example of what that might look like in this use case? This is my first foray into zod.

Reply:
Mmmmh, I'm also a newbie to zod. My assumption was that this could work the same as with union & intersection types in TS.

I'm just [reading](https://github.com/colinhacks/zod?tab=readme-ov-file#or) that `z.or` is basically a convenience method for union types ^^ (same goes for `and` and `intersection`)

So far, I imagine this should be defined like follows without a custom refinement:

```
z.object({
  // Shared properties
  displayName: z.string().optional(),
  description: z.string().optional(),
  group: z.string().optional(),
  // TODO: add validations as defined in ComponentSettingsSchema
}).and(z.union([
  z.object({
    type: z.literal('Array'),
    defaultValue: z.any().array().optional(),
  }),
  z.object({
    type: z.literal('Boolean'),
    defaultValue: z.boolean().optional(),
  }),
  // ... same for the remaining cases
])
```

Reply:
I think you're PR is already a great improvement, so feel free to skip this. Maybe, we come back to this static validation approach when the need for it arises. So far, the only direct advantage would be more readable types created ouf of zod - something that would be okay to be addressed later as well

Reply:
I gave this a try, and while it worked, I was having a hard time getting the error messages to be useful. For some reason passing in `message` wasn't being used. 

I agree on this approach is better for typing and code readability, but going to go with this solution for now to get it done.